### PR TITLE
Add a few more precompiles

### PR DESCRIPTION
**Pull request by timholy**
These precompiles help ImageCore precompile other MethodInstances.

### CODE REVIEW

File: src/precompile.jl [hunk 1]
```diff
@@ -59,6 +59,8 @@ function _precompile_()
     end
     # traits
     for T in eltypes, C in pctypes
+        @assert precompile(Tuple{typeof(eltype),Type{C}})
+        @assert precompile(Tuple{typeof(eltype),Type{C{T}}})
         @assert precompile(Tuple{typeof(alpha),C{T}})
         C <: AbstractGray && @assert precompile(Tuple{typeof(gray),C{T}})
         C <: AbstractRGB || continue
```
Comment (kimikage) [inline src/precompile.jl:66]: This is not directly related to this PR, but are the `red()`, `green()` and `blue()` for `TransparentRGB` not precompiled?

**Review by kimikage — APPROVED** [review]
I haven't quantified the benefits, but at least the precompilation of ColorTypes isn't too slow.

File: src/precompile.jl (hunk 1 above)
Comment (timholy) [reply src/precompile.jl:66]: No, I could do that but I think transparent colors are less commonly-used. And of course having them for `RGB` saves some time when it does go to compile them.

**Comment by timholy** [comment]
Benefits are not huge but it helps other packages. There's a chain from ColorTypes->ImageCore->PNGFiles, and `load(filename)` takes 0.2s to infer which is a substantial time.
